examples/main/main.cpp

```diff
@@ -215,24 +237,27 @@ int main(int argc, char ** argv) {
 
     if (!path_session.empty()) {
         LOG_TEE("%s: attempting to load saved session from '%s'\n", __func__, path_session.c_str());
-
-        // fopen to check for existing session
-        FILE * fp = std::fopen(path_session.c_str(), "rb");
-        if (fp != NULL) {
-            std::fclose(fp);
-
+        if (!file_exists(path_session))
+        {
+            LOG_TEE("%s: session file does not exist, will create.\n", __func__);
+        }
+        else if (file_size(path_session) == 0)
+        {
+            LOG_TEE("%s: The session file is empty. A new session will be initialized.\n", __func__);
+        }
+        else
+        {
+            // The file exists and is not empty
             session_tokens.resize(n_ctx);
             size_t n_token_count_out = 0;
-            if (!llama_load_session_file(ctx, path_session.c_str(), session_tokens.data(), session_tokens.capacity(), &n_token_count_out)) {
+            if (!llama_load_session_file(ctx, path_session.c_str(), session_tokens.data(), session_tokens.capacity(), &n_token_count_out))
+            {
```

Comment:
Opening braces on same line:

```suggestion
            if (!llama_load_session_file(ctx, path_session.c_str(), session_tokens.data(), session_tokens.capacity(), &n_token_count_out)) {
```

Reply:
thanks, formatted: 0fcdefc346d584d05c4cc7486bc87cc02b097763